crates/rooch-types/src/transaction/authenticator.rs

```diff
@@ -187,59 +114,49 @@ where
     T: BuiltinAuthenticator,
 {
     fn from(value: T) -> Self {
-        let scheme = value.scheme() as u64;
+        let coin_id = value.coin_id() as u64;
         let payload = value.payload();
-        Authenticator { scheme, payload }
+        Authenticator { coin_id, payload }
     }
 }
 
 impl From<Signature> for Authenticator {
-    fn from(sign: Signature) -> Self {
-        match sign.to_public_key().unwrap().scheme() {
-            BuiltinScheme::Ed25519 => Authenticator::ed25519(sign),
-            BuiltinScheme::Ecdsa => Authenticator::ecdsa(sign),
-            BuiltinScheme::EcdsaRecoverable => Authenticator::ecdsa_recoverable(sign),
-            BuiltinScheme::MultiEd25519 => todo!(),
-            BuiltinScheme::Schnorr => Authenticator::schnorr(sign),
-        }
+    fn from(signature: Signature) -> Self {
+        Authenticator::rooch(signature)
+    }
+}
+
+impl From<Secp256k1RecoverableSignature> for Authenticator {
+    fn from(ethereum_signature: Secp256k1RecoverableSignature) -> Self {
+        Authenticator::ethereum(ethereum_signature)
     }
 }
 
 #[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
 pub struct Authenticator {
-    pub scheme: u64,
+    pub coin_id: u64,
     pub payload: Vec<u8>,
 }
 
 impl Authenticator {
     /// Unique identifier for the signature scheme
-    pub fn scheme(&self) -> u64 {
-        self.scheme
+    pub fn coin_id(&self) -> u64 {
```

Comment:
The coin_id here is a bit strange. Maybe confuse the user with the coin in the `coin.move`. But we can record an issue to discuss later.

@baichuan3 @wow-sven

Reply:
`multichain_id` is fit cuz there's a `chain_id` in the genesis context.

Reply:
I recorded an issue:

https://github.com/rooch-network/rooch/issues/791

I think we can just use custom chain id to build each blockchain ids by hand.